Library/ViewModels/CommentsViewModel.swift

```diff
@@ -216,13 +216,30 @@ public final class CommentsViewModel: CommentsViewModelType,
 
     let commentComposerDidSubmitText = self.commentComposerDidSubmitTextProperty.signal.skipNil()
 
-    self.failableOrComment <~ Signal.combineLatest(
-      initialProject,
-      currentUser.skipNil()
-    )
-    .takePairWhen(commentComposerDidSubmitText)
-    .map(unpack)
-    .flatMap(.concurrent(limit: concurrentCommentLimit), postCommentProducer)
+    let commentableId = projectOrUpdate
+      .flatMap { projectOrUpdate in
+        projectOrUpdate.ifLeft { project in
+          SignalProducer.init(value: project.graphID)
+        } ifRight: { update in
+          SignalProducer.init(value: encodeToBase64("FreeformPost-\(update.id)"))
+        }
+      }
+
+    let postFailableCommentConfigData:
+      Signal<(project: Project, commentableId: String, user: User), Never> = Signal.combineLatest(
+        initialProject,
+        commentableId,
+        currentUser.skipNil()
+      ).map { configData in
```

Comment:
replace with:

`.map { project, commentableId, user in
(project: project, commentableId: commentableId, user: user)
}` for readability.